include/pstore/mcrepo/linked_definitions_section.hpp

```diff
@@ -79,12 +90,37 @@ namespace pstore {
         /// record.
         class linked_definitions : public section_base {
         public:
-            using iterator = typed_address<compilation_member> *;
-            using const_iterator = typed_address<compilation_member> const *;
-
-            template <typename Iterator, typename = typename std::enable_if<std::is_same<
-                                             typename std::iterator_traits<Iterator>::value_type,
-                                             typed_address<compilation_member>>::value>::type>
+            struct value_type {
```

Comment:
Could you add some documents for the struct of value_type?

Reply:
Done in commit 0dd1b4e3d4d7e6d64d124aef28faef9e677a4c73.